components/collector/src/source_collectors/dependency_track/source_up_to_dateness.py

```diff
@@ -87,3 +91,21 @@ def _timestamp_to_datetime(timestamp: int | None) -> datetime | None:
     def _is_latest(project: DependencyTrackProject) -> bool:
         """Return whether the project is the latest version."""
         return project.get("isLatest", False)
+
+    def _up_to_dateness(self, project: DependencyTrackProject) -> str:
+        """Return the up-to-dateness of the project."""
+        date_times = self._project_event_datetimes(project)
+        if not date_times:
+            return "unknown"
+        min_date_time = self.days(self.minimum(date_times))
+        if self._target_met(min_date_time, "target"):
+            return "yes"
+        if self._target_met(min_date_time, "near_target"):
+            return "almost"
+        return "no"
+
+    def _target_met(self, value: int, target_type: Literal["target", "near_target"]) -> bool:
+        """Return whether the value meets the target value."""
+        target = int(self._metric.get(target_type) or getattr(DATA_MODEL.metrics[self._metric["type"]], target_type))
+        direction = str(self._metric.get("direction") or DATA_MODEL.metrics[self._metric["type"]].direction.value)
+        return value <= target if direction == "<" else value >= target
```

Comment:
If for some reason the up to dateness could have a direction `>`, shouldn't this also invert the min (or max) selector on line 100?

Reply:
Good point. Until now, source collectors didn't have access to the metric parameters, but now they do, so may just as well use that in the `minimum()` method.